src/python/WMCore/Services/LogDB/LogDBBackend.py

```diff
@@ -86,13 +96,35 @@ def prefix(self, mtype):
             mtype = 'agent-%s' % mtype
         return mtype
 
-    def post(self, request, msg='', mtype="comment"):
-        """Post new entry into LogDB for given request"""
+    def agent_update(self, request, msg='', mtype="comment"):
+        """Update agent info in LogDB for given request"""
         self.check(request, mtype)
         mtype = self.prefix(mtype)
-        data = {"request":request, "agent": self.dbid, "worker": self.thread_name,
-                "ts":tstamp(), "msg":msg, "type":mtype}
-        res = self.db.commitOne(data)
+        data = {"ts":tstamp(), "msg":msg, "type":mtype}
+        doc = {"_id": self.docid(request, mtype), "data": [data]}
+        try:
+            exist_doc = self.db.document(doc["_id"])
+            doc["_rev"] = exist_doc["_rev"]
+        except CouchNotFoundError:
+            # this means document is not exist so we will just insert
+            pass
+        finally:
+            res = self.db.commitOne(doc)
+        return res
+
+    def user_update(self, request, msg, mtype='comment'):
+        """Update user info in LogDB for given request"""
+        data = {"ts":tstamp(), "msg":msg, "type":mtype}
+        doc = {"_id": self.docid(request, mtype), "data": [data]}
```

Comment:
In case we have view by type, it might be better to take the type out from data since type will be only 'comment'.
Or are you thinking about extending to other message type?

```
data = {"ts":tstamp(), "msg":msg}
 doc = {"_id": self.docid(request, mtype), "comments": [data], "type": "comment"}
```